File: 0011-container-with-most-water/0011-container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int first = 0 , end = height.Length - 1;
                int max = int.MinValue ;
                while(first < end)
                {
                    int width = end - first;
                    int heigth = Math.Min(height[end], height[first]);
                    max = Math.Max(max, (width  * heigth));
                    if (height[end] >= height[first]) 
                        first++;
                    
                    else
                    
                        end--;
                    
                }
                return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0, j = height.length - 1;
        int area = -1;
        while (i < j){
            area = Math.max(area, Math.min(height[i],height[j]) * (j-i));
            if(height[i] < height[j])
                i++;
            else j--;
        }
        return area;
    }
}